src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -479,38 +477,33 @@ private String evaluateMacro(String template) {
 
     private void pullRequestActionable(List<PotentialAction> paList, List<Fact> factsList) {
         Job job = run.getParent();
-        ItemGroup parent = job.getParent();
-        if (parent instanceof MultiBranchProject) {
-            BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
-            if (projectFactory.isProject(job)) {
-                Branch branch = projectFactory.getBranch(job);
-                SCMHead head = branch.getHead();
-
-                if (head instanceof ChangeRequestSCMHead) {
-                    String pronoun = StringUtils.defaultIfBlank(head.getPronoun(), "Change Request");
-                    String viewName = String.format("View %s", pronoun);
-                    String titleName = String.format("%s Title", pronoun);
-                    String authorName = String.format("%s Author", pronoun);
-
-                    ObjectMetadataAction oma = branch.getAction(ObjectMetadataAction.class);
-                    if (oma != null) {
-                        String urlString = oma.getObjectUrl();
-                        PotentialAction viewPRPotentialAction = new PotentialAction(viewName, urlString);
-                        paList.add(viewPRPotentialAction);
-                        factsList.add(new Fact(titleName, oma.getObjectDisplayName()));
-                    }
-                    ContributorMetadataAction cma = branch.getAction(ContributorMetadataAction.class);
-                    if (cma != null) {
-                        String contributor = cma.getContributor();
-                        String contributorDisplayName = cma.getContributorDisplayName();
-                        String author = StringUtils.defaultIfBlank(cma.getContributor(), cma.getContributorDisplayName());
-                        if (StringUtils.isNotBlank(contributor) && StringUtils.isNotBlank(contributorDisplayName))
-                            author = String.format("%s (%s)", cma.getContributor(), cma.getContributorDisplayName());
-
-                        if (StringUtils.isNotBlank(author))
-                            factsList.add(new Fact(authorName, author));
-                    }
-                }
+        SCMHead head = SCMHead.HeadByItem.findHead(job);
+        if (head instanceof ChangeRequestSCMHead) {
+            String pronoun = StringUtils.defaultIfBlank(
```

Comment:
Don't you need to check SCMHead for null?

Reply:
`instanceof` is checking for null
https://stackoverflow.com/questions/2950319/is-null-check-needed-before-calling-instanceof

Read the discussion
https://github.com/jenkinsci/office-365-connector-plugin/pull/22#discussion_r143955230

We are correcting a bad use of Branch API and it's is not needed.

Reply:
Removing all this unnecessary code is not seen as a gain?
```
        ItemGroup parent = job.getParent();
        if (parent instanceof MultiBranchProject) {
            BranchProjectFactory projectFactory = ((MultiBranchProject) parent).getProjectFactory();
            if (projectFactory.isProject(job)) {
                Branch branch = projectFactory.getBranch(job);
                SCMHead head = branch.getHead();
```